src/main/cpp/src/hive_hash.cu

```diff
@@ -210,11 +208,228 @@ class hive_device_row_hasher {
       return this->hash_functor.template operator()<T>(col, row_index);
     }
 
+    /**
+     * @brief A structure representing an element in the stack used for processing columns.
+     *
+     * This structure is used to keep track of the current column being processed, the index of the
+     * next child column to process, and current hash value for the column.
+     *
+     * @param _column The current column being processed.
+     * @param _child_idx The index of the next child column to process.
+     * @param _cur_hash The current hash value for the column.
+     *
+     * @note The default constructor is deleted to prevent uninitialized usage.
+     *
+     * @constructor
+     * @param col The column device view to be processed.
+     */
+    class col_stack_element {
+     private:
+      cudf::column_device_view _column;  // current column
```

Comment:
Instead of storing a column in each stack element, you can first flatten the input nested column into a table (array of columns), then here just store the index of the column in that array. See `cudf::experimental::row::lexicographic::preprocessed_table` for example of such flatten table.

Some preprocessing is needed so we can retrieve the index of the children columns in the array of flatten table.

Reply:
Check out `cudf/cpp/src/table/row_operators.cu` (`cudf::experimental::row::lexicographic::preprocessed_table::create`) and `cudf/cpp/src/structs/utilities.cpp` (`cudf::structs::detail::flatten_nested_columns`) for example code of flattening table.

Reply:
This approach may require significant changes to the existing framework, as the calculations for these 5 columns are not independent.